transformations/random_walk/transformation.py

```diff
@@ -0,0 +1,215 @@
+        tokenizer ([type]): tokenizer
+        model ([type]): model
+        k (int, optional): how many replacement words to try. Defaults to 5.
+
+    Returns:
+        [list]: k-sentences with masked word replaced with top-k most similar words
+    """
+    #text_split = re.split('[ ?.,!;"]', sentence)
+    split_iter = re.finditer(r"[\w']+|[.,!?;]", sentence)
+    text_split = []
+    split_indices = []
+    for m in split_iter:
+      text_split.append(m.group(0))
+      split_indices.append((m.start(), m.end()))
+
+    included_mask = False
+    while not included_mask: # Loop until the masked word is one of the first 512 tokens of input, since all else are discarded.
+      # pick a random word to mask
+      
+      rand_int = np.random.randint(len(text_split))
+      word_to_mask = text_split[rand_int]
+      while len(word_to_mask) == 0 and not word_to_mask.isalnum(): # Avoid empty strings in split text
+          rand_int = np.random.randint(len(text_split))
+          word_to_mask = text_split[rand_int]
+      
+      # mask word
+      new_text = _mask_word(sentence, split_indices[rand_int], tokenizer.mask_token)
+
+      # get k replacement words
+      top_k, included_mask = get_k_replacement_words(new_text, tokenizer, model, k=k)
+    replacement_words = [tokenizer.decode([token]) for token in top_k]
+
+    # replace mask-token with the word from the top-k replacements
+    return [
+        new_text.replace(tokenizer.mask_token, word)
+        for word in replacement_words
+    ]
+
+
+def single_round(sentences: List[str], tokenizer, model, k) -> List[str]:
+    """For a given list of sentences, do a random walk on each sentence.
+
+    Args:
+        sentences ([type]): list of sentnces to perform random walk on
+        tokenizer ([type]): tokenizer
+        model ([type]): model
+
+    Returns:
+        [List]: list of random-walked sentences
+    """
+    new_sentences = []
+
+    for sentence in sentences:
+        new_sentences.extend(
+            single_sentence_random_step(sentence, tokenizer, model, k))
+
+    return new_sentences
+
+
+def random_walk(original_text: str, steps: int, k: int, tokenizer,
+                model) -> List[str]:
+    old_sentences = [original_text]
+
+    # Do k steps of random walk procedure
+    for _ in range(steps):
+        sentences = single_round(old_sentences, tokenizer, model, k)
+        old_sentences = copy.deepcopy(sentences)
+
+    #assert len(sentences) == k**steps  # This may not be possible if we cannot find k non-punctuation suggestions.
+    return sentences
+
+def sentence_similarity_metric(similarity_model, sen_A, sen_B):
+    emb_A = similarity_model.encode(sen_A)
+    emb_B = similarity_model.encode(sen_B)
+
+    score = util.pytorch_cos_sim(emb_A, emb_B)
+    return score
+
+class RandomWalk(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_CLASSIFICATION
+    ]
+    languages = ["en"]
+
```

Comment:
don't forget to include keywords: https://github.com/GEM-benchmark/NL-Augmenter/blob/main/docs/keywords.md 
like:
`keywords = [
        "model-based", "lexical", "possible-meaning-alteration",
        "high-coverage", "high-generations"
    ]`